sources/packages/backend/apps/queue-consumers/src/processors/schedulers/base-scheduler.ts

```diff
@@ -0,0 +1,40 @@
+import { OnApplicationBootstrap } from "@nestjs/common";
+import Bull, { CronRepeatOptions, Queue } from "bull";
+
+export abstract class BaseScheduler<T> implements OnApplicationBootstrap {
+  abstract initializeScheduler(): void;
+  abstract cronOptions: Bull.JobOptions;
+  abstract schedulerQueue: Queue<T>;
+
+  /**
+   * Once all modules have been initialized
+   * it will check, if there is any old cron job
+   * delete it and add the new job to the
+   * queue.
+   */
+  async onApplicationBootstrap() {
```

Comment:
Can we change it to `protected`?

Reply:
![image](https://user-images.githubusercontent.com/77353155/205165838-4b0fce2b-bfd7-49d8-a68f-dbb4564a2149.png)